### PR TITLE
Plot astropix hits in real time

### PR DESCRIPTION
**Pull request by henrikef**
Add a module to plot and display astroPix hits (row and column) in real time.

For now, run `plotDemo.py` to see a demo of the GUI.

Interface to the data collection script is still missing.

### CODE REVIEW

**Comment by c235labtop** [comment]
[AS] Tested on lab computer - looks great! 

- Maybe additionally create a directory to save every red/orange image as a PDF? 
- (0,0) pixel in bottom left (I know... but Manoj confirmed)

@Narg3000 's updates are merged to main so you have access to data in lists now

**Comment by henrikef** [comment]
Saving the images does drive up the deadtime, which means there is less time to wait for signals from astroPix. That's why I didn't include the option originally. It's added now, please check on the lab computer to make sure it runs under windows and the dead time is not too high. For reference, I was seeing around 70 ms before I added the save statement. With this version, the deadtime increased to ~350 ms for the first event that had a file to be saved, and ~100 ms for subsequent red/orange events.

File: beam_test.py [hunk 1]
```diff
@@ -175,12 +179,29 @@ def main():
             decode.decode_astropix2_hits(decode.hits_from_readoutstream(readout), i, file1)
             #decode.decode_astropix2_hits(decode.hits_from_readoutstream(readout))
             file1.write("\n")
+            
+            rows = [] #placeholder for list of rows hit
```
Comment (henrikef) [inline beam_test.py]: @asteinhebel @Narg3000 you will need to change this line and the following line. The `plot_event` function takes two lists, the IDs of the triggering rows and columns. These can be lists or numpy arrays. Here I just put empty lists as placeholders.

**Comment by henrikef** [comment]
@asteinhebel the calling signature changed, you will now need to specify `-n` or `--name` on the command line if you want to provide an extra string for the file name. `-s` to display hits in real-time (once the lists are implemented) and `-o` to set and output directory to save hit displays.

**Comment by c235labtop** [comment]
[AS] - tested on lab computer - saving plots also shows average deadtime of 0.1s. WIP to test with running chip

**Comment by c235labtop** [comment]
[AS]  Tested on lab computer and in workable state after commits

To summarize commits:
- return row/col as bool 0/1 for easier management
- implement -s runtime option
- use decoder optional argument to return list of decoded values
- use decoded values as input for plotter
- failsafe requiring decoded hits to exist before considering them as input - otherwise input empty arrays to plotter

**Pull request merged**